agent/agent.go

```diff
@@ -316,10 +320,14 @@ func shouldRunSetupGroup(nextTask *apimodels.NextTaskResponse, tc *taskContext)
 				"next_task_task_group":    nextTask.TaskGroup,
 			})
 		}
-		return true
+		msg = "running setup group because new task group"
+		setupGroup = true
 	}
 
-	return false
+	if tc.logger != nil {
```

Comment:
depending on how actively you want to monitor this, you can also log to the agent's log file if the task logger isn't setup